clients/cli/connections.c

```diff
@@ -7764,10 +7771,20 @@ editor_menu_main (NmCli *nmc, NMConnection *connection, const char *connection_t
 					update_connection (persistent, rem_con, update_connection_editor_cb, NULL);
 				}
 
-				//FIXME: add also a timeout for cases the callback is not called
+				connection_changed = FALSE;
+				handler_id = g_signal_connect (rem_con,
+				                               NM_CONNECTION_CHANGED,
+				                               G_CALLBACK (editor_connection_changed_cb),
+				                               &connection_changed);
+
 				while (!nmc_editor_cb_called)
 					g_main_context_iteration (NULL, TRUE);
 
+				while (!connection_changed)
```

Comment:
if the signal for some reason does not come, this loops indefinitely. I think we need a timeout here, maybe 5 seconds?

The rest lgtm.